src/main/webapp/app/course/manage/detail/course-detail-line-chart.component.html

```diff
@@ -1,24 +1,44 @@
-<div class="col">
+<div class="course-detail-parent-container col">
     <h5 class="col-md-12 text-center" style="margin-top: 15px">{{ 'artemisApp.courseStatistics.activeStudents' | artemisTranslate }}</h5>
     <div class="row flex-nowrap" style="display: flex; justify-content: center; align-items: center; flex: 1">
         <fa-icon [icon]="'arrow-left'" size="2x" class="col-md-1 d-flex justify-content-end align-items-center" role="button" (click)="switchTimeSpan(LEFT)"></fa-icon>
-        <div class="col-md-9">
-            <canvas
-                class="course-detail-line-chart"
-                baseChart
-                height="750"
-                width="1500"
-                [datasets]="chartData"
-                [labels]="lineChartLabels"
-                [options]="lineChartOptions"
-                [legend]="lineChartLegend"
-                [chartType]="lineChartType"
-            ></canvas>
+        <div #containerRef class="chart col-md-9 mt-1 justify-content-center h-99">
+            <ngx-charts-line-chart
+                [view]="[containerRef.offsetWidth, 450]"
+                [scheme]="chartColor"
+                [legend]="legend"
+                [showXAxisLabel]="showXAxisLabel"
+                [showYAxisLabel]="showYAxisLabel"
+                [xAxis]="xAxis"
+                [yAxis]="yAxis"
+                [xAxisLabel]="xAxisLabel"
+                [yAxisLabel]="yAxisLabel"
+                [timeline]="timeline"
+                [results]="data"
+                [yScaleMin]="0"
+                [yScaleMax]="100"
+                [yAxisTickFormatting]="formatYAxis"
+            >
+                <ng-template #tooltipTemplate let-model="model">
```

Comment:
#tooltipTemplate and #seriesTooltipTemplate both have no usages locally. Are they needed?

Reply:
@julian-christl Thnaks for applying the latest guidelines to my tests :) And yes, ngx-charts uses those two templates to create the tooltips of the chart.
- `#tooltipTemplate` declares the tooltip when hovering over a data point
- `#seriestooltipTemplate` declares the tooltip when hovering over any place in the graph (here it's just a small tooltip)
To my knowledge @willergeorg will create a guide on how to work with ngx-charts.